lib/Sema/CSSimplify.cpp

```diff
@@ -1208,6 +1208,17 @@ static bool allowsBridgingFromObjC(TypeChecker &tc, DeclContext *dc,
   return true;
 }
 
+/// Determine whether the given type variables occurs in the given type.
+static bool typeVarOccursInType(TypeVariableType *typeVar, Type type) {
```

Comment:
The solver doesn't need to get the representative (that's just wasted cycles), but it tracks that other bit 'involves other type variables'. I can make that last parameter optional to unify them. Thanks!